sol/main.c

```diff
@@ -504,14 +544,45 @@ static int
 sol_stage1(void *arg)
 {
 	struct sol_config *sol_conf = arg;
-	int ret = get_queue_id(&sol_conf->net->back, QUEUE_TYPE_TX,
-		sol_conf->lcore_id, NULL);
-	if (ret < 0) {
-		SOL_LOG(ERR, "Cannot assign a TX queue for the back interface for lcore %u\n",
-			sol_conf->lcore_id);
+	int ret, i;
+
+	sol_conf->instances = rte_calloc_socket(__func__, sol_conf->num_lcores,
+		sizeof(struct sol_instance), 0,
+		rte_lcore_to_socket_id(sol_conf->lcores[0]));
+	if (sol_conf->instances == NULL)
 		goto cleanup;
+
+	for (i = 0; i < sol_conf->num_lcores; i++) {
+		unsigned int lcore = sol_conf->lcores[i];
+		struct sol_instance *inst_ptr = &sol_conf->instances[i];
+		char ring_name[64];
+
+		ret = snprintf(ring_name, sizeof(ring_name), "sol_reqs_ring_%u",
+			i);
+		RTE_VERIFY(ret > 0 && ret < (int)sizeof(ring_name));
+
+		inst_ptr->ring = rte_ring_create(ring_name,
+			rte_align32pow2(sol_conf->pri_req_max_len),
+			rte_lcore_to_socket_id(lcore), RING_F_SC_DEQ);
+		if (inst_ptr->ring == NULL) {
+			G_LOG(ERR,
+				"sol: can't create ring sol_reqs_ring at lcore %u\n",
+				lcore);
+			ret = -1;
```

Comment:
This assignment of variable `ret` is not being used. In fact, the declaration of `ret` should be done within the `for` loop since it's only used inside of the loop.

Reply:
You've forgotten to drop the line `ret = -1;` shown above.

Reply:
Thanks! Done.